assets/app/styles/_core.less

```diff
@@ -343,6 +343,9 @@ label.checkbox {
     }
     .help-block {
       margin-bottom: 5px;
+      a {
+        cursor: pointer;
```

Comment:
You know I was going crazy trying to find the style that caused the problem. It turns out it was because there is no `href` on the link. Will fix.